src/transformers/pipelines/zero_shot_image_classification.py

```diff
@@ -0,0 +1,125 @@
+    `"image-classification"`.
+
+    See the list of available models on
+    [huggingface.co/models](https://huggingface.co/models?filter=image-classification).
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        if self.framework != "pt":
+            raise ValueError(f"The {self.__class__} is only available in PyTorch.")
+
+        requires_backends(self, "vision")
+        # No specific FOR_XXX available yet
+        # self.check_model_type(MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING)
+
+    def __call__(self, images: Union[str, List[str], "Image", List["Image"]], **kwargs):
+        """
+        Assign labels to the image(s) passed as inputs.
+
+        Args:
+            images (`str`, `List[str]`, `PIL.Image` or `List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing a http link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+                The pipeline accepts either a single image or a batch of images, which must then be passed as a string.
+                Images in a batch must all be in the same format: all as http links, all as local paths, or all as PIL
+                images.
+            candidate_labels (`List[str]`):
+                The candidate labels for this image
+            hypothesis_template (`str`, *optional*, defaults to `"This is a photo of a {}"`):
+                The sentence used in cunjunction with *candidate_labels* to attempt the image classification by
+                replacing the placeholder with the candidate_labels. Then likelihood is estimated by using
+                likelihood_per_image
+
+        Return:
+            A dictionary or a list of dictionaries containing result. If the input is a single image, will return a
+            dictionary, if the input is a list of several images, will return a list of dictionaries corresponding to
+            the images.
+
+            The dictionaries contain the following keys:
+
+            - **label** (`str`) -- The label identified by the model.
+            - **score** (`int`) -- The score attributed by the model for that label.
+        """
+        return super().__call__(images, **kwargs)
+
+    def _sanitize_parameters(self, **kwargs):
+        preprocess_params = {}
+        if "candidate_labels" in kwargs:
+            preprocess_params["candidate_labels"] = kwargs["candidate_labels"]
+        if "hypothesis_template" in kwargs:
+            preprocess_params["hypothesis_template"] = kwargs["hypothesis_template"]
+
+        postprocess_params = {}
+        if "multi_label" in kwargs:
+            postprocess_params["multi_label"] = kwargs["multi_label"]
+        return preprocess_params, {}, postprocess_params
+
+    def preprocess(self, image, candidate_labels=None, hypothesis_template="This is a photo of {}."):
```

Comment:
It needs to be sent as a parameter since that was how it was done in  `zero-shot-classification`.
(debatable choice, but it does enable `pipe = pipeline(..., candidate_labels["hot-dog", "pizza"]), pipe(image_dataset)`)

Parameters can't be `[]` by default (first no mutable default arguments, second it does not make sense to run the pipeline on an empty set of  candidates.

2 solutions to improve the error when user forgets candidate_labels: (current it will be something along the  llines that None is not iterable)

- add `if candidate_labels is None: raise Exception("BETTER EXCEPTION MESSAGE")`. simple, I like it.
- change signature to `preprocess(self, image, *, candidate_labels,...)`. I need to check, but I think that will give a nice error message directly sent by Python.

Reply:
> but it does enable pipe = pipeline(..., candidate_labels["hot-dog", "pizza"]), pipe(image_dataset))

I see. Thanks for the explanation. Then in this case I would also prefer the first option.